library/src/main/java/com/readystatesoftware/chuck/internal/data/repository/ChuckerDatabaseRepository.kt

```diff
@@ -0,0 +1,67 @@
+package com.readystatesoftware.chuck.internal.data.repository
+
+import android.arch.lifecycle.LiveData
+import com.readystatesoftware.chuck.internal.data.entity.HttpTransaction
+import com.readystatesoftware.chuck.internal.data.entity.HttpTransactionTuple
+import com.readystatesoftware.chuck.internal.data.entity.RecordedThrowable
+import com.readystatesoftware.chuck.internal.data.entity.RecordedThrowableTuple
+import com.readystatesoftware.chuck.internal.data.room.ChuckerDatabase
+import java.util.concurrent.Executor
+import java.util.concurrent.Executors
+
+internal class ChuckerDatabaseRepository(private val database: ChuckerDatabase) : ChuckerRepository {
+
+    override fun getFilteredTransactionTuples(code: String, path: String): LiveData<List<HttpTransactionTuple>> {
+        val pathQuery = if (path.isNotEmpty()) "%$path%" else "%"
+        return database.transactionDao().getFilteredTransactionsTuples("$code%", pathQuery)
+    }
+
+    override fun getTransaction(transactionId: Long): LiveData<HttpTransaction> {
+        return database.transactionDao().getRecordedTransaction(transactionId)
+    }
+
+    override fun getSortedTransactionTuples(): LiveData<List<HttpTransactionTuple>> {
+        return database.transactionDao().getSortedTransactionsTuples()
+    }
+
+    override fun deleteAllTransactions() {
+        executor.execute { database.transactionDao().deleteAllTransactions() }
+    }
+
+    override fun getRecordedThrowable(id: Long): LiveData<RecordedThrowable> {
+        return database.throwableDao().getRecordedThrowable(id)
+    }
+
+    override fun deleteAllThrowables() {
+        executor.execute { database.throwableDao().deleteAllThrowables() }
+    }
+
+    override fun getSortedThrowablesTuples(): LiveData<List<RecordedThrowableTuple>> {
+        return database.throwableDao().getSortedThrowablesTuples()
+    }
+
+    private val executor: Executor = Executors.newSingleThreadExecutor()
+
+    override fun insertTransaction(transaction: HttpTransaction) {
```

Comment:
I'm not really fan of this kind of method that have a side effect (here, changing `transaction.id`). For later we should think about it.

Reply:
Wasn't really able to find better alternatives. Here the problem is that the `id` field is auto-generated and we should update the fields. AFAIK, the auto-generated id is returned as result value of the `insert` function.